### ISSUE TITLE
MIDI file playing too slow

### ISSUE THREAD

**Issue by h3x4d3c1m4l** [atsushieno/managed-midi#14]
When playing flourish.mid (Windows sample MIDI file), the file seems to play with wrong bpm. It even seems that it does not play the file with constant bpm.

I'm running below sample code on .NET 4.5 on Windows 10 x64 on a very recent Core i7 CPU. The CPU usage is very low while playing.

```
var access = MidiAccessManager.Default;
var output = access.OpenOutputAsync(access.Outputs.Last().Id).Result;
var music = MidiMusic.Read(System.IO.File.OpenRead(@"C:\windows\media\flourish.mid"));
var player = new MidiPlayer(music, output);
player.PlayAsync();
```

**Comment by atsushieno** [comment]
https://github.com/atsushieno/managed-midi/blob/master/lib/base/MidiTimeManager.cs#L57 - Maybe Task.Delay() behaves super slow on Windows. A "precise" time manager that absorbs ideal and actual time might "fix" the issue. Time-preserving time manager one of my TODO ideas, but that may conflict with support for tempo changes.

**Comment by MikeTheWatchGuy** [comment]
Does non "Timer-preserving" time manager mean you drift over time so that if a MIDI reports lasting 3:21 you may actually take 3:25?

**Comment by atsushieno** [comment]
That sounds too much. I made some change in the default time manager to try to fix this potential problem at 9a19990 but there still seems to be some delays. I'll take more look in depth once I got free time later this week (I'm super busy preparing local conference talk these days).

**Comment by atsushieno** [comment]
I had a song file which is of 5 min. 35 sec. long, and my MIDI player https://github.com/atsushieno/xmdsp always showed like 5:37 at the end, so there was definitely some delays. Now with cce1998 it exactly stops at 5:35. I take it as this issue is finally fixed.

If you guys still see delays please let me know here. Thanks for reporting the issue!
